cmd/topicctl/subcmd/rebalance.go

```diff
@@ -151,55 +159,98 @@ func rebalanceRun(cmd *cobra.Command, args []string) error {
+	tmpDir := ""
+	existingConfigFiles := make(map[string]struct{})
+	if rebalanceConfig.generateConfig {
+		// make set of existing files
+		err := processTopicFiles(topicFiles, func(topicConfig config.TopicConfig, topicFile string) error {
+			existingConfigFiles[topicConfig.Meta.Name] = struct{}{}
+			return nil
+		})
 		if err != nil {
-			log.Errorf("Invalid topic yaml file: %s", topicFile)
-			continue
+			return err
 		}
 
-		for _, topicConfig := range topicConfigs {
-			// topic config should be consistent with the cluster config
-			if err := config.CheckConsistency(topicConfig.Meta, clusterConfig); err != nil {
-				log.Errorf("topic file: %s inconsistent with cluster: %s", topicFile, clusterConfigPath)
-				continue
-			}
+		// create temp dir
+		tmpDir = rebalanceConfig.pathPrefix
+		if rebalanceConfig.pathPrefix[len(rebalanceConfig.pathPrefix) - 1] != '/' {
+			tmpDir += "/"
+		}
+		tmpDir += "tmp/"
+		err = os.Mkdir(tmpDir, 0755)
+		if err != nil {
+			return err
+		}
+		log.Infof("tmp output path: %v", tmpDir)
+
+		// generate (bootstrap) config files
+		cliRunner := cli.NewCLIRunner(adminClient, log.Infof, false)
+		cliRunner.BootstrapTopics(
+			ctx,
+			[]string{},
+			clusterConfig,
+			".*",
+			".^",
+			tmpDir,
+			false,
+			false,
+		)
 
-			log.Infof(
-				"Rebalancing topic %s from config file %s with cluster config %s",
-				topicConfig.Meta.Name,
-				topicFile,
-				clusterConfigPath,
-			)
-
-			topicErrorDict[topicConfig.Meta.Name] = nil
-			rebalanceTopicProgressConfig := util.RebalanceTopicProgressConfig{
-				TopicName:          topicConfig.Meta.Name,
-				ClusterName:        clusterConfig.Meta.Name,
-				ClusterEnvironment: clusterConfig.Meta.Environment,
-				ToRemove:           rebalanceConfig.brokersToRemove,
-				RebalanceError:     false,
-			}
-			if err := rebalanceApplyTopic(ctx, topicConfig, clusterConfig, adminClient); err != nil {
-				topicErrorDict[topicConfig.Meta.Name] = err
-				rebalanceTopicProgressConfig.RebalanceError = true
-				log.Errorf("topic: %s rebalance failed with error: %v", topicConfig.Meta.Name, err)
-			}
+		// re-invetory config files to take into account newly generated ones
+		topicFiles, err = getAllFiles(topicConfigDir)
+		if err != nil {
+			return err
+		}
+	}
 
-			// show topic final progress
-			if rebalanceCtxStruct.Enabled {
-				progressStr, err := util.StructToStr(rebalanceTopicProgressConfig)
-				if err != nil {
-					log.Errorf("progress struct to string error: %+v", err)
-				} else {
-					log.Infof("Rebalance Progress: %s", progressStr)
-				}
+	// iterate through each topic config and initiate rebalance
+	topicErrorDict := make(map[string]error)
+	processed := processTopicFiles(topicFiles, func(topicConfig config.TopicConfig, topicFile string) error {
+		// skip generated config if it already existed
+		if rebalanceConfig.generateConfig && topicConfigExists(topicFile, existingConfigFiles, topicConfig.Meta.Name) {
```

Comment:
I think the logs might be more intuitive if we prune these out before we start the rebalancing. Otherwise it's going to be unclear why we are skipping some things